### DISCUSSION TITLE
Xdrip Backfill in Companion Mode

### DISCUSSION THREAD

**Discussion by ghost** [NightscoutFoundation/xDrip#2426] [Q&A]
Hi,
Somehow the backfill in companion mode seems not working for me. I am using CamAPS fx and Xdrip in companion mode.
I have activated the new companion Bluetooth Settings.

Do I still have to activate something?

thanks in advance

Daniel 

![90323F5C-AAD2-4B38-AF2C-7ED1CBF27523](https://user-images.githubusercontent.com/59935735/199759685-2172680e-d96a-4891-aa9c-07601f716f33.png)
![0C6E1F0F-1F08-4C2D-A2A0-E747E2EBFC2E](https://user-images.githubusercontent.com/59935735/199759694-d79fccd6-d562-4f9e-9c68-b3952d242e50.png)

**Comment by AliQ101** [comment]
still not realiable #2008

**Comment by jamorham** [comment]
@AliQ101 @1987Daniel Can you both try the Nov 4th Nightly to see if it resolves the issue. thanks.

**Reply by AliQ101** [reply]
@jamorham - thanks for your hard work. I tried Nov 4th Nightly (for G6 BYODA) and no progress for me, no backfills in companion mode with companion bluetooth on. Compared to Oct 15th Nightly there are any entries in log such as "Encountered backfill data..." no more, jut only "Did no find any matches", which seems strange.
![Screenshot_20221105-151842](https://user-images.githubusercontent.com/107270312/200128202-a3cf2bc4-d05d-4a6f-aac4-46375ff78dd8.png)
![Screenshot_20221105-151912](https://user-images.githubusercontent.com/107270312/200128206-fbe9e109-2581-4b26-8637-709bd47e03e0.png)

**Comment by jamorham** [comment]
@AliQ101 I think I will need more information to resolve this. To assist me, please follow these instructions. Go to `Settings` -> `Less common settings` -> `Extra Logging Settings` and select `Extra tags for logging` and then enter exactly `ClassifierAction:d` and let this run for say an hour, then create a situation where the transmitter is away from the phone, then after say 15 minutes, return within range so that a backfill should be generated. When you see that this has happened, please then use the `Upload Logs` feature from the `Events Log` screen and make sure you enter your github name in somewhere so I can find it and then let me know here when that is done. This should give me more technical information to try to diagnose why its not working for you.

**Reply by AliQ101** [reply]
@jamorham I have just sent the Logs.

**Reply by jamorham** [reply]
@AliQ101 @1987Daniel  Thank you AliQ for the logs. These show me something but really it is the lack of data which is significant. I need more information again to see what is going on. Can you again set the `Extra tags for logging` as before but this time add on `,Bluetails:d` so it reads `ClassifierAction:d,Bluetails:d` and repeat the tests as best you can and send me logs again when it should have gathered enough log data.

**Reply by AliQ101** [reply]
@jamorham I have just the Logs again (14.13).

**Reply by ghost** [reply]
@jamorham I have sent the Logs today (10:42)

**Reply by jamorham** [reply]
@1987Daniel It looked like your backfill at approx 10:30 worked fine. Was there still a problem?

**Reply by ghost** [reply]
@jamorham thank you for your answer. on the graph from yesterday i don't see any values from backfill at 10:30. see attached screenshot
![Screenshot_20221108-050428](https://user-images.githubusercontent.com/59935735/200492994-2f71bc9f-d460-4cbe-9b6f-7443d91621b7.png)

**Reply by PieterDeBruijn** [reply]
Sent in my logs too.

**Reply by jamorham** [reply]
@1987Daniel Please try the Nov 8th Nightly, it is designed to address specifically your issue that I have seen from your logs.
@AliQ101 please keep sending logs. I think your problem is different, it may be that this technique is only reliable on newer android versions - I don't know yet.
@PieterDeBruijn Your logs didn't seem to have the additional log tags enabled as I described above so there was limited information. Please also try Nov 8th Nightly in case it is the same issue I have already seen.

**Reply by ghost** [reply]
@jamorham with the nightly from 8th November the backfill works for me. Many thanks

**Reply by AliQ101** [reply]
@jamorham I sent the logs (for 8th Nov Nightly) - for me it seems that companion mode with BT off is more stable (just like 3 Aug version) compared to 14 and 15 Oct, 4 and 5 Nov versions (there were accidental losses of reading when G6 transmitter was in range). With BT on, it seems that there were made 2 backfills, majority was missed however.

**Reply by jamorham** [reply]
@AliQ101 I think your problem is that this technique doesn't work well on older android versions. I believe you use android 9. I think it is possible it may only work properly on android 12 and above but I don't have enough people testing it to know for sure.

@1987Daniel great news that this is now working.

@PieterDeBruijn Daniel is reporting success now, could you try with the latest nightly and if it isn't working, please advise which android version you are using and/or submit logs which include the extra log tags I described previously that need to be enabled.

**Reply by AliQ101** [reply]
@jamorham  yes, your're right, Im on Android 9. Unfortunately I have no phone with Android 10 or 11,but I found one with 12. I will use it for testing companion BT purposes to see if it is also stable for me.

**Reply by PieterDeBruijn** [reply]
Backfill working properly, reporting from Android 12 (Pixel 4 XL). Will test the Android 13 update, as soon as CamAPS has official support.

**Reply by jamorham** [reply]
I think this is resolved except the technique may not work on older android versions or may be related to specific bluetooth implementations on certain handsets. We would need more testers to get a better idea.

**Comment by ghost** [comment]
@jamorham Thank you for your work. I have tested the nightly from 4th November. Also for me it is the same that the backfill does not work. I try to follow your instruction also for more details.

**Reply by Navid200** [reply]
Please do that for the latest, meaning November 5, 2022.
